File: leaderboards/app/database/Parser.java
package database;

import models.Leaderboard;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Date;

public class Parser {
    public static Leaderboard toLeaderboard(ResultSet resultSet) throws SQLException {
        int gameId = resultSet.getInt(Structure.Leaderboards.Columns.GameId);
        String leaderboardName = resultSet.getString(Structure.Leaderboards.Columns.LeaderboardName);

        long startTime = resultSet.getLong(Structure.Leaderboards.Columns.StartTime);
        long endTime = resultSet.getLong(Structure.Leaderboards.Columns.EndTime);

        boolean descending = resultSet.getInt(Structure.Leaderboards.Columns.Descending) == 1;

        return new Leaderboard(
                gameId,
                leaderboardName,
                startTime == -1 ? null : new Date(startTime),
                endTime == -1 ? null : new Date(endTime),
                descending
        );
    }
}

File: authentication/app/database/Parser.scala
package database

import java.sql.ResultSet
import database.models._

object Parser {
  def toApp(resultSet: ResultSet) {
    new App(
      resultSet.getInt(Structure.Apps.Columns.Id),
      resultSet.getString(Structure.Apps.Columns.AppIdentifier),
      resultSet.getString(Structure.Apps.Columns.FullName),
      resultSet.getString(Structure.Apps.Columns.Description)
    )
  }

  def toGroup(resultSet: ResultSet) {
    new Group(
      resultSet.getInt(Structure.Groups.Columns.Id),
      resultSet.getString(Structure.Groups.Columns.Name),
      resultSet.getString(Structure.Groups.Columns.Description)
    )
  }

  def toPermission(resultSet: ResultSet) {
    new Permission(
      resultSet.getInt(Structure.Permissions.Columns.Id),
      resultSet.getString(Structure.Permissions.Columns.Name),
      resultSet.getString(Structure.Permissions.Columns.Description)
    )
  }

  def toUser(resultSet: ResultSet) {
    new User(
      resultSet.getInt(Structure.Users.Columns.Id),
      resultSet.getString(Structure.Users.Columns.Name),
      resultSet.getString(Structure.Users.Columns.Email),
      resultSet.getBytes(Structure.Users.Columns.Password).toList,
      resultSet.getBytes(Structure.Users.Columns.Salt).toList,
      resultSet.getString(Structure.Users.Columns.ApiKey),
      resultSet.getDate(Structure.Users.Columns.CreatedOn),
      resultSet.getInt(Structure.Users.Columns.FailedAttempts),
      resultSet.getBoolean(Structure.Users.Columns.Blocked)
    )
  }
}